modules/pipeline/main.tf

```diff
@@ -0,0 +1,76 @@
+resource "aws_lambda_function" "cuddle_serverless" {
+  # If the file is not in the current working directory you will need to include a
+  # path.module in the filename.
+  filename      = "lambda_function_payload.zip"
```

Comment:
if I understood correctly, this is a module that you are calling in the main file of the project let's say
but, you are passing static values, which interferes with the concept of the module.
a module should be global, generic, and has no static values (use variables instead) to be able to use it multiple times